### PR TITLE
TST Add test for loc on sparse dataframes

### LINKED ISSUE #34687: BUG: multiple application of .loc on sparse DataFrame results in NaN and filling the DataFrame
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
# Your code here

from scipy.sparse import coo_matrix, eye  
import pandas as pd

df = pd.DataFrame.sparse.from_spmatrix(eye(10))

df.sparse.density 
# Should output 0.1

df.loc[range(5)]                                                        
#     0    1    2    3    4    5    6    7    8    9
# 0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
# 1  0.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
# 2  0.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
# 3  0.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0
# 4  0.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0

df.loc[range(5)].sparse.density
# outputs 0.1

df.loc[range(5)].loc[range(3)]                                          
#     0    1    2    3    4   5   6   7   8   9
# 0  1.0  0.0  0.0  0.0  0.0 NaN NaN NaN NaN NaN
# 1  0.0  1.0  0.0  0.0  0.0 NaN NaN NaN NaN NaN
# 2  0.0  0.0  1.0  0.0  0.0 NaN NaN NaN NaN NaN

df.loc[range(5)].loc[range(3)].sparse.density
# outputs 0.6

```

#### Problem description

It seems that sparse DataFrame extracted using loc does not behave as the original ones. This creates inconsistencies in our processing pipelines, depending on filtering and selection that has been applied, sometimes producing "Nans", severely impacting memory consumption and computational time. 

#### Expected Output

The output of loc should not depend on multiple slicing, i.e.

df.loc[range(5)].loc[range(3)] = df.loc[range(3)]

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None

pandas           : 1.0.4
numpy            : 1.18.5
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 9.0.3
setuptools       : 39.0.1
Cython           : 0.29.19
pytest           : 5.4.3
hypothesis       : 5.16.1
sphinx           : 3.1.0
blosc            : 1.9.1
feather          : None
xlsxwriter       : 1.2.9
lxml.etree       : 4.5.1
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.15.0
pandas_datareader: None
bs4              : 4.9.1
bottleneck       : 1.3.2
fastparquet      : 0.4.0
gcsfs            : None
lxml.etree       : 4.5.1
matplotlib       : 3.2.1
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : 0.17.1
pytables         : None
pytest           : 5.4.3
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.4.1
sqlalchemy       : 1.3.17
tables           : 3.6.1
tabulate         : 0.8.7
xarray           : 0.15.1
xlrd             : 1.2.0
xlwt             : 1.3.0
xlsxwriter       : 1.2.9
numba            : 0.49.1

</details>

### PR DESCRIPTION
**Pull request by EricLeer**
Adds a test for loc on sparse dataframes.

- [x] closes #34687
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

### CODE REVIEW

**Comment by EricLeer** [comment]
The test `pandas/tests/io/xml/test_xml.py::test_parser_consistency_url` seems to be failing, but I am not seeing how I am touching this code by just adding another test. Does anyone know how I would be able to fix this?

**Comment by jbrockmendel** [comment]
> The test pandas/tests/io/xml/test_xml.py::test_parser_consistency_url seems to be failing, but I am not seeing how I am touching this code by just adding another test. Does anyone know how I would be able to fix this?

This is unrelated and OK to ignore here.  Just restarted that build.

File: pandas/tests/indexing/test_loc.py [hunk 1]
```diff
@@ -1163,6 +1163,25 @@ def test_loc_getitem_listlike_all_retains_sparse(self):
         result = df.loc[[0, 1]]
         tm.assert_frame_equal(result, df)
 
+    @td.skip_if_no_scipy
+    def test_loc_sparse_frame(self):
```
Comment (jbrockmendel) [inline pandas/tests/indexing/test_loc.py:1167]: can you start the name with `test_loc_getitem` to match the pattern in this clas

mainly this is testing that .loc indexing retains SparseDtype right?  should test for Series too?

Comment (EricLeer) [reply pandas/tests/indexing/test_loc.py]: Have updated the name, and also added similar tests for sparse series, or would it be better to brake these out into a seperate function?

Comment (jreback) [reply pandas/tests/indexing/test_loc.py]: yeah i think if you can move the series tests to series/indexing/test_indexing

Comment (EricLeer) [reply pandas/tests/indexing/test_loc.py]: I have moved them to that folder.

**Comment by jreback** [comment]
@EricLeer the pre-commit is failing if you can have a look

File: pandas/tests/indexing/test_loc.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/indexing/test_loc.py]: sorry fellas, rule of thumb: if you're testing .loc, it belongs in tests.indexing.test_loc.  See https://github.com/pandas-dev/pandas/blob/master/doc/source/development/test_writing.rst

Comment (EricLeer) [reply pandas/tests/indexing/test_loc.py]: No problem, have now moved it back to `tests.indexing.test_loc`.

**Comment by jreback** [comment]
thanks @EricLeer

**Pull request merged**